source/topic5.rst

```diff
@@ -14,9 +14,85 @@ Example Uses
 Interface
 =========
 
+* There are many possible ways one could implement a stack data structure
+* But, all implementations must be a *stack*
+    * They must follow our definition of a stack ADT
+
+* In Java, we can create an **interface** that defines what the operations of the stack ADT are without actually implementing it
+* Then, when someone implements the stack ADT, the interface will make sure that the implementation *implements* the operators
+
+* The interface will simply be a list of *abstract methods* and relevant constants
+    * Abstract methods are the method signature/header and no actual body
+    * Our methods will be ``public``
+        * ``public int someMethod(int a, int b);``
+    * Constants will be ``static final``
+
+StackADT Interface
+------------------
+
+.. code-block:: java
+    :linenos:
+
+    public interface StackADT <T> {
+
+        public void push(T element);
+        public T pop();
+        public T peek();
+        public boolean isEmpty();
+        public int size();
+        public String toString();
+
+    }
+
+
 Generics
 --------
 
+* You are probably wondering what ``<T>`` is
+
+* Imagine you wanted to have a stack of type ``Integer``, then you'd have to make
+    * ``public push(int element);``
+    * ``public int pop();``
```

Comment:
A `Stack<Integer>` does not an `IntStack` make. `Integer` is **not** an `int`. `int` -> `Integer` here.